src/metabase/driver/sql_jdbc/actions.clj

```diff
@@ -14,29 +14,32 @@
             [metabase.query-processor.store :as qp.store]
             [metabase.util :as u]
             [metabase.util.honeysql-extensions :as hx]
-            [metabase.util.i18n :refer [tru]]
+            [metabase.util.i18n :refer [tru trs]]
             [toucan.db :as db])
   (:import java.sql.Connection))
 
 (defmulti parse-sql-error
   "Parses the raw error message returned after an error in the driver database occurs, and converts it into a sequence
   of maps with a :column and :message key indicating what went wrong."
-  {:arglists '([driver ^java.sql.Connection connection e]), :added "0.44.0"}
+  {:arglists '([driver database e]), :added "0.44.0"}
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
 
 (defn- parse-error
   "Returns errors in a way that indicates which column had the problem. Can be used to highlight errors in forms."
-  [driver conn e]
+  [driver database e]
   (let [default-method? (= (get-method parse-sql-error driver)
                            (get-method parse-sql-error :default))]
     (if default-method?
       (ex-data e)
       (let [message (ex-message e)]
-        (if-let [errors (and message
-                             (parse-sql-error driver conn message))]
-
-          {:errors (->> errors
+        (if-let [parsed-errors (when message
+                                 (try
+                                   (parse-sql-error driver database message)
+                                   (catch Throwable e
+                                     (log/error e (trs "Error parsing SQL error message {0}: {1}" (pr-str message) (ex-message e)))
+                                     nil)))]
```

Comment:
Catch errors in `parse-sql-error` and log them so more errors in the future don't break the entire action. We'll still get the original unparsed error message.